surfer/tests/test_viz.py

```diff
@@ -113,21 +113,22 @@ def test_brains():
 @requires_fsaverage
 def test_annot():
     """Test plotting of annot."""
-    mlab.options.backend = 'test'
+    mlab.options.backend = 'auto'
```

Comment:
Yes, it's a different problem – if I don't do it it looks like new arrays are not assigned to the data-source, I get:
```
$ nosetests test_viz:test_annot
2017-06-14 15:49:44.261 python[48695:54025567] modalSession has been exited prematurely - check for a reentrant call to endModalSession:
Exception occurred in traits notification handler for object: <mayavi.tools.filters.SetActiveAttributeFactory object at 0x13166a950>, trait: point_scalars, old value: , new value: 2
Traceback (most recent call last):
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler( *args )
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 198, in _anytrait_changed
    setattr(obj, components[-1], value)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/core/trait_defs.py", line 97, in set_value
    raise TraitError(object, name,"one of %s"%values, value)
TraitError: The 'point_scalars_name' trait of a SetActiveAttribute instance must be one of [], but a value of '2' <type 'str'> was specified.
E
======================================================================
ERROR: Test plotting of annot.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/Users/christian/Code/PySurfer/surfer/tests/test_viz.py", line 122, in test_annot
    brain.add_annotation(a, b, p)
  File "/Users/christian/Code/PySurfer/surfer/viz.py", line 1229, in add_annotation
    brain['brain'].add_annotation(annot, ids, cmap))
  File "/Users/christian/Code/PySurfer/surfer/viz.py", line 2830, in add_annotation
    array_id, pipe = self._add_scalar_data(ids)
  File "/Users/christian/Code/PySurfer/surfer/viz.py", line 2727, in _add_scalar_data
    self._mesh_dataset, point_scalars=array_id, figure=self._f)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 38, in the_function
    factory = factory_class(*args, **kwargs)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 163, in __init__
    self.set(**traits)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 180, in set
    self._anytrait_changed(trait, value)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 198, in _anytrait_changed
    setattr(obj, components[-1], value)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/core/trait_defs.py", line 97, in set_value
    raise TraitError(object, name,"one of %s"%values, value)
TraitError: The 'point_vectors_name' trait of a SetActiveAttribute instance must be one of [], but a value of '' <type 'str'> was specified.
-------------------- >> begin captured stdout << ---------------------
Exception occurred in traits notification handler.
Please check the log file for details.

--------------------- >> end captured stdout << ----------------------
-------------------- >> begin captured logging << --------------------
apptools.preferences.preferences: DEBUG: loading preferences from </Users/christian/.enthought/mayavi_e3/preferences.ini>
apptools.preferences.preferences: DEBUG: loading preferences from <<open file '/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/preferences/preferences.ini', mode 'rb' at 0x11da33ed0>>
apptools.preferences.preferences: DEBUG: loading preferences from <<open file '/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/tvtk/plugins/scene/preferences.ini', mode 'rb' at 0x11db008a0>>
mayavi.core.registry: DEBUG: Engine [<mayavi.core.null_engine.NullEngine object at 0x128e8ac50>] named NullEngine1 registered
mayavi.core.common: ERROR: No valid current object, please select an active object.
traits: ERROR: Exception occurred in traits notification handler for object: <mayavi.tools.filters.SetActiveAttributeFactory object at 0x13166a950>, trait: point_scalars, old value: , new value: 2
Traceback (most recent call last):
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/traits/trait_notifiers.py", line 340, in __call__
    self.handler( *args )
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/tools/pipe_base.py", line 198, in _anytrait_changed
    setattr(obj, components[-1], value)
  File "/Users/christian/anaconda/envs/mayavi/lib/python2.7/site-packages/mayavi/core/trait_defs.py", line 97, in set_value
    raise TraitError(object, name,"one of %s"%values, value)
TraitError: The 'point_scalars_name' trait of a SetActiveAttribute instance must be one of [], but a value of '2' <type 'str'> was specified.
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 2.321s

FAILED (errors=1)
```